include/deal.II/lac/cuda_kernels.templates.h

```diff
@@ -238,8 +206,15 @@ namespace LinearAlgebra
             __syncthreads();
           }
 
-        if (local_idx < 32)
-          reduce_within_warp<Number, Operation>(result_buffer, local_idx);
+        if (local_idx < warp_size)
```

Comment:
I am not sure if you want to replace the unrolled version with a loop... NVIDIA normally recommends to completely unroll it...

Reply:
Yeah but how do I unroll by hand if I don't know the size. The loop size is known at compile time so I would expect the compiler to be able to do it.